packages/test/jest.test.js

```diff
@@ -0,0 +1,171 @@
+/* eslint-env node, jest */
+
+const hrp = require("http-request-plus").default;
+const jrp = require("json-rpc-protocol");
+const keyBy = require("lodash/keyBy");
+
+async function call(method, params) {
+  return jrp.parse.result(
+    await hrp
+      .post("http://localhost:4000/api/", {
+        body: jrp.format.request(0, method, params),
+      })
+      .readAll("utf-8")
+  );
+}
+
+// swap promise fulfilment/rejection, very handy to test rejections
+const invertPromise = p =>
+  p.then(
+    value => {
+      throw value;
+    },
+    reason => reason
+  );
+
+function compareTimestamps(ts1, ts2) {
+  return expect(Math.abs(ts1 - ts2)).toBeLessThan(1e2);
+}
+
+let entries, id1, id2, id3;
+
+test("create entry", async () => {
+  id1 = await call("createEntry", { name: "name1", content: "content1" });
+  id2 = await call("createEntry", { content: "content2" });
+  id3 = await call("createEntry", { name: "name3" });
+
+  const now = Date.now();
+  const response = await call("listEntries");
+
+  expect(response).toEqual([
+    {
+      id: id1,
+      name: "name1",
+      content: "content1",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+    {
+      id: id2,
+      content: "content2",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+    {
+      id: id3,
+      name: "name3",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+  ]);
+
+  entries = keyBy(response, "id");
+
+  keyBy(response, function(entry) {
+    expect(entry.created).toBe(entry.updated);
+    compareTimestamps(entry.created, now);
+  });
+});
+
+test("update entry", async () => {
+  await call("updateEntry", {
+    id: id1,
+    name: "name1_modified",
+    content: "content1_modified",
+  });
+  await call("updateEntry", { id: id2, name: "only_name2_modified" });
+  await call("updateEntry", {
+    id: id3,
+    content: "only_content3_modified",
+  });
+
+  const now = Date.now();
+  const response = await call("listEntries");
+
+  expect(response).toEqual([
+    {
+      id: id1,
+      name: "name1_modified",
+      content: "content1_modified",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+    {
+      id: id2,
+      name: "only_name2_modified",
+      content: "content2",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+    {
+      id: id3,
+      name: "name3",
+      content: "only_content3_modified",
+      created: expect.any(Number),
+      updated: expect.any(Number),
+    },
+  ]);
+
+  const entriesUpdated = keyBy(response, "id");
+  expect(entriesUpdated[id1].created).toBe(entries[id1].created);
+  expect(entriesUpdated[id2].created).toBe(entries[id2].created);
+  expect(entriesUpdated[id3].created).toBe(entries[id3].created);
+
+  keyBy(response, entry => compareTimestamps(entry.updated, now));
```

Comment:
That's not the use case for `keyBy`.

Reply:
You could use [`Array#forEach()`](https://devdocs.io/javascript/global_objects/array/foreach).